x-pack/plugins/security_solution/public/resolver/data_access_layer/mocks/emptify_mock.ts

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import {
+  ResolverRelatedEvents,
+  ResolverTree,
+  ResolverEntityIndex,
+} from '../../../../common/endpoint/types';
+import { mockTreeWithNoProcessEvents } from '../../mocks/resolver_tree';
+import { DataAccessLayer } from '../../types';
+
+type EmptiableRequests = 'relatedEvents' | 'resolverTree' | 'entities' | 'indexPatterns';
+
+interface Metadata<T> {
+  /**
+   * The `_id` of the document being analyzed.
+   */
+  databaseDocumentID: string;
+  /**
+   * A record of entityIDs to be used in tests assertions.
+   */
+  entityIDs: T;
+}
+
+/**
+ * A simple mock dataAccessLayer that allows you to control whether a request comes back with data or empty.
+ */
+export function emptifyMock<T>(
+  {
+    metadata,
+    dataAccessLayer,
+  }: {
+    dataAccessLayer: DataAccessLayer;
+    metadata: Metadata<T>;
+  },
+  dataShouldBeEmpty?: EmptiableRequests[]
```

Comment:
Does this need to be optional? If you don't pass it then there would be no point in using `emptifyMock` at all, right?